src/app/problems/_forms/java.form.ts

```diff
@@ -19,6 +19,7 @@ export class JavaForm {
             text: new FormControl('', [Validators.required]),
             junit_template: new FormControl(null, [Validators.required]),
             input_files: new FormControl([], [Validators.required]),
+            variation_types: new FormControl(['No Variations']),
```

Comment:
did you add `variation_types` to the question model? We need to get this field to show it  when editing the problems. Make sure editing is also working properly.